prql-compiler/src/ast/pl/ident.rs

```diff
@@ -1,136 +1,96 @@
+        if self.path().is_empty() {
+            (self.name(), None)
         } else {
-            let first = self.path.remove(0);
-            (first, Some(self))
+            let first = self.0.remove(0);
+            (first, Some(Ident(self.0)))
         }
     }
-
-    pub fn with_name<S: ToString>(mut self, name: S) -> Self {
-        self.name = name.to_string();
-        self
-    }
-
-    fn into_vec(self) -> Vec<String> {
-        self.path.into_iter().chain(Some(self.name)).collect()
-    }
-
     pub fn starts_with(&self, prefix: &Ident) -> bool {
-        self.clone()
-            .into_vec()
-            .starts_with(&prefix.clone().into_vec())
-    }
-}
-
-#[test]
-fn test_starts_with() {
-    // Over-testing, from co-pilot, can remove some of them.
-    let a = Ident::from_path(vec!["a", "b", "c"]);
-    let b = Ident::from_path(vec!["a", "b"]);
-    let c = Ident::from_path(vec!["a", "b", "c", "d"]);
-    let d = Ident::from_path(vec!["a", "b", "d"]);
-    let e = Ident::from_path(vec!["a", "c"]);
-    let f = Ident::from_path(vec!["b", "c"]);
-    assert!(a.starts_with(&b));
-    assert!(a.starts_with(&a));
-    assert!(!a.starts_with(&c));
-    assert!(!a.starts_with(&d));
-    assert!(!a.starts_with(&e));
-    assert!(!a.starts_with(&f));
-}
-
-impl std::fmt::Display for Ident {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        display_ident(f, self)
-    }
-}
-
-impl IntoIterator for Ident {
-    type Item = String;
-    type IntoIter = std::iter::Chain<
-        std::vec::IntoIter<std::string::String>,
-        std::option::IntoIter<std::string::String>,
-    >;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.path.into_iter().chain(Some(self.name))
+        self.0.starts_with(&prefix.0)
     }
 }
 
 impl std::ops::Add<Ident> for Ident {
     type Output = Ident;
 
     fn add(self, rhs: Ident) -> Self::Output {
-        Ident {
-            path: self.into_iter().chain(rhs.path).collect(),
-            name: rhs.name,
-        }
-    }
-}
-
-impl Serialize for Ident {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        let mut seq = serializer.serialize_seq(Some(self.path.len() + 1))?;
-        for part in &self.path {
-            seq.serialize_element(part)?;
-        }
-        seq.serialize_element(&self.name)?;
-        seq.end()
-    }
-}
-
-impl<'de> Deserialize<'de> for Ident {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        <Vec<String> as Deserialize>::deserialize(deserializer).map(Ident::from_path)
+        Ident(self.0.into_iter().chain(rhs.0).collect())
     }
 }
 
-pub fn display_ident(f: &mut std::fmt::Formatter, ident: &Ident) -> Result<(), std::fmt::Error> {
-    for part in &ident.path {
+// Q: @aljazerzen do we need these now? I couldn't immediately see what they
+// do (but can look more if needed).
```

Comment:
No, we don't. This makes it so the Ident was serialized into an array, even though it was a struct. But now this should have no effect, because it's gets serialized into array anyway.